pkg/diagnostics/utils/test_utils.go

```diff
@@ -0,0 +1,32 @@
+package utils
```

Comment:
This file should probably be called `utils_test.go` and/or have a `//go:build test` tag, so it's only built in tests.

Otherwise, the runtime itself will then depend on the testing package

Reply:
agree, good spot!

Reply:
Ok so it's not possible (as far as I can tell) to import code from a test package (either defined by the `_test.go` file name or a build tag) so I moved it into an isolated `testutils` package which is only imported via the test code so won't be included in the final `daprd` binary.

Reply:
Uhmmm I thought I got that to work in the past.

Anyways, with the last change, do we even need a separate package now? Or could that be put in the test files directly. Seems to be used by a single package?

Reply:
it's used by 2 tests in a the `diagnostics` package. I either copy the function to both test files or I include it in the package which then gets bundled into the `daprd` binary which was your original objection. Putting it in `testutils` is only to ensure it is not imported in any non test code.

Reply:
Include it in the `diagnostics` package and make sure the file is called `_test.go`. Then it will be only included in test code, and can be used by any (test) file in the `diagnostics` package

Reply:
ah right, gotcha, I tried that via a build tag and it didn't appear to work but the file name does seem to work, thanks! Updated.